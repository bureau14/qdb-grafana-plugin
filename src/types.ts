import { DataQuery, DataSourceJsonData } from '@grafana/data';

export interface MyQuery extends DataQuery {
  queryText?: string;
  constant: number;
}

export const defaultQuery: Partial<MyQuery> = {
  constant: 6.5,
};

/**
 * Value that is used in the backend, but never sent over HTTP to the frontend
 */
export interface MySecureJsonData {
  apiKey?: string;
}

/**
 * These are options configured for each DataSource instance
 */
export interface MyDataSourceOptions extends DataSourceJsonData {
  uri?: string;
}

/**
 * Value that is used in the backend, but never sent over HTTP to the frontend
 */
export interface SecureJsonData {
  username?: string;
  user_private_key?: string;
  cluster_public_key?: string;
}
