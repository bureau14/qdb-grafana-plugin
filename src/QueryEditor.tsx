import defaults from 'lodash/defaults';

import React, { ChangeEvent, PureComponent } from 'react';
import { TextArea, Button } from '@grafana/ui';
import { QueryEditorProps } from '@grafana/data';
import { DataSource } from './datasource';
import { defaultQuery, QdbDataSourceOptions, QdbQuery } from './types';

type Props = QueryEditorProps<DataSource, QdbQuery, QdbDataSourceOptions>;

export class QueryEditor extends PureComponent<Props> {
  onQueryTextChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
    const { onChange, query } = this.props;
    onChange({ ...query, queryText: event.target.value });
  };

  render() {
    const query = defaults(this.props.query, defaultQuery);
    const { queryText } = query;

    return (
      <div className="gf-form">
        <TextArea label="Query" value={queryText || ''} onChange={this.onQueryTextChange}></TextArea>
        <Button onClick={this.props.onRunQuery}>Run Query</Button>
      </div>
    );
  }
}
