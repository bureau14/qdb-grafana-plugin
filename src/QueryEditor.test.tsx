// QueryEditor.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { QdbQuery } from './types';
import { QueryEditor } from './QueryEditor';

test('QueryEditor', () => {
  let query: QdbQuery = {
    refId: 'aaa',
    queryText: 'SELECT * FROM ts;',
    tagQuery: true,
  };
  const component = renderer.create(<QueryEditor query={query}></QueryEditor>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
