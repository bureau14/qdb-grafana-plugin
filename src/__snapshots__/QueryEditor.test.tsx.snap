// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryEditor 1`] = `
<div
  className="gf-form"
>
  <textarea
    className="css-1f38lle"
    label="Query"
    onChange={[Function]}
    value="SELECT * FROM ts;"
  />
  <button
    className="css-a5ta8x-button"
    onClick={[MockFunction]}
  >
    <span
      className="css-1mhnkuh"
    >
      Run Query
    </span>
  </button>
</div>
`;
